main.py

```diff
@@ -37,6 +88,135 @@ def _return_model(model_name: str) -> BaseModel:
+            return Bar
+        case 'waterfall':
+            return Waterfall
+        case 'mesh3d':
+            return Mesh3d
+        case 'barpolar':
+            return Barpolar
+        case 'scattersmith': 
+            return Scattersmith
+        case 'streamtube':
+            return Streamtube
+        case 'cone': 
+            return Cone
+        case 'scattermapbox':
+            return Scattermapbox
+        case 'scattergeo': 
+            return Scattergeo
+        case 'scatterpolar':
+            return Scatterpolar
+        case 'sunburst':
+            return Sunburst
+        case 'histogram2d':
+            return Histogram2d
+        case 'isosurface':
+            return Isosurface
+        case 'violin':
+            return Violin
+        case 'image':
+            return Image
+        case 'ohlc':
+            return Ohlc
+        case 'heatmapgl':
+            return Heatmapgl
+        case 'indicator':
+            return Indicator
+        case 'funnelarea':
+            return Funnelarea
+        case 'carpet': 
+            return Carpet
+        case 'icicle':
+            return Icicle
+        case 'surface':
+            return Surface
+        case 'parcats': 
+            return Parcats
+        case 'treemap':
+            return Treemap
+        case 'funnel':
+            return Funnel
+        case 'histogram2dcontour':
+            return Histogram2dcontour
+        case 'contourcarpet':
+            return Contourcarpet
+        case 'parcoords':
+            return Parcoords
+        case 'candlestick':
+            return Candlestick
+        case 'scatter3d':
+            return Scatter3d
+        case 'choropleth':
+            return Choropleth
+        case 'heatmap':
+            return Heatmap
+        case 'histogram': 
+            return Histogram
+        case 'volume':
+            return Volume
+        case 'contour':
+            return Contour
+        case 'scatterternary':
+            return Scatterternary
+        case 'sankey':
+            return Sankey
+        case 'scattercarpet':
+            return Scattercarpet
+        case 'densitymapbox':
+            return Densitymapbox
+        case 'choroplethmapbox':
+            return Choroplethmapbox
+        case 'box':
+            return Box
+        case 'pie':
+            return Pie
+        case 'scatterpolargl':
+            return Scatterpolargl
+        case 'scattergl':
+            return Scattergl
+        case 'splom':
+            return Splom
+        
+def is_pydantic_model(cls) -> bool:
+    """Check if a given class is a Pydantic model."""
+    return isinstance(cls, type) and issubclass(cls, BaseModel)
+
+def extract_model_info(model: Type[BaseModel]) -> Dict[str, Any]:
+    output = {}
+
+    for field_name, field_info in model.model_fields.items():
+        field_type = field_info.annotation
+        field_description = '' if field_info.description == None else ' #' + field_info.description
```

Comment:
I did write that. I left it out of the final string that goes into the marco because it made the yaml file unreadable. 

We definitely need to find a way to add the description. I was thinking maybe leveraging the markdown table of contents would be a good way to do that like you're suggesting, but just wanted to get this pass pushed and come back to it another PR.

Reply:
@tgsoverly